src/modules/formBuilder/components/itemEditor/conditionals/EnableWhenCondition.tsx

```diff
@@ -163,6 +163,22 @@ const EnableWhenCondition: React.FC<EnableWhenConditionProps> = ({
 
   const localConstantsPickList = useLocalConstantsPickList();
 
+  const clearAnswer = useCallback(() => {
+    setValue(`${enableWhenPath}.${index}.answerCode`, null);
+    setValue(`${enableWhenPath}.${index}.answerCodes`, null);
+    setValue(`${enableWhenPath}.${index}.answerBoolean`, null);
+    setValue(`${enableWhenPath}.${index}.answerNumber`, null);
+    setValue(`${enableWhenPath}.${index}.answerGroupCode`, null);
+  }, [setValue, enableWhenPath, index]);
+
+  const onChangeDependentQuestion = useCallback(() => {
+    // todo @Martha - fix typescript's unhappiness
```

Comment:
Ok so: What I found when playing around with this is that:
- `shouldUnregister` appears to be the culprit for the bug that's video'd in the original ticket. I wonder if it is because `shouldUnregister` causes a reset on the form, which resets to `defaultValues`?
- `shouldUnregister` also doesn't appear to be working -- at least, in the "answer value" dropdowns. Using the debug button, I could see that even when the dropdown or text box component unmounted, the `answerCode` or `answerBoolean` property of the `enableWhen` would still be present.
- The code as it is here appears to me to be working, but typescript is unhappy with passing `null` to the enableWhen operator. I've added a couple ideas for how to move forward, @gigxz I'd like to get your thoughts and/or pair :)

Reply:
I did some digging here to try to understand _why_ the components were being unregistered when changing/clearing the value.  I pasted the below code in various components in this hierarchy to see which ones were being mounted and unmounted
```
  useEffect(() => {
    console.log('parent component mounted');
    return () => {
      console.log('parent component unmounted');
    };
  }, []);
```

That tracked me back to several keys [like this](https://github.com/greenriver/hmis-frontend/blob/3298215d9ffb94ba22357409d87df5fb9e0d1934/src/modules/formBuilder/components/itemEditor/conditionals/ManageEnableWhen.tsx#L91) (I'm sorry 🙈 ). Those keys were causing the whole component tree below them to mount unmount when values changed. The unmounting caused the inputs to unregister, and the remounting caused them to reregister with their default values.

I opened https://github.com/greenriver/hmis-frontend/pull/906, can you try that out and see if it addresses the bug from the ticket? I think it should, but I only did some basic testing.

Also, I had no idea that useFieldArray generated `id`s.. I found that out from seeing that you used it on the PickListOption array. So thanks!